### ISSUE TITLE
Simplify magic.node.extensions

### ISSUE THREAD

**Issue by polterguy** [polterguy/magic#153]
This project works, but its code base is a mess, especially the parts related to Hyperlambda parsing/generating, and the project needs to be seriously cleaned up, refactored, and improved upon. The project has a lot of unit test coverage though, so this shouldn't be a problem.

What we'll need is basically two different functions;

* The ability to parse Hyperlambda
* The ability to create Hyperlambda from a list of Nodes

The parser needs to be able to take a Stream input, in addition to an overload taking a string. The thing needs to be `async`, since sometimes we're probably going to read Hyperlambda directly from files, and/or HTTP invocations, etc - Hence, async support is crucial.

The code needs to be **significantly** easier to read and follow, and less complex than its current implementation, and well documented, and all existing unit tests needs to pass - And the end result should have close to 100% coverage, something I can check once the code is merged into the master branch by using SonarCloud.

The API can be changed from its current implementation, as long as it has all the existing abilities it currently have, which includes a boolean value declaring whether or not we're supposed to parse comments or not.

If comments are parsed, these should end up like a **[..]** node, with the value of the comment being the value of the node, minus any spacings not considered a part of the comment.

The project needs to take any arbitrary piece of Hyperlambda, parse it to a node, and reverse the process, resulting in one ending up with the same Hyperlambda as one started out with, assuming comments were semantically correctly applied in the original Hyperlambda.

The result also needs to be at least somewhat close to its current performance, hence a unit test which loads up a ton of Hyperlambda files, hundreds of times, measuring its performance both before and after, showing how the project is still (at least) on pair with the current execution speed needs to be created.

Notice, the end resulting code must tolerate all different permutations of `\n`, `\r\n` and `\r` as newline characters, to make sure we support all OS CR/LF sequences.
